src/components/Sidebar/ChannelsSidebar/Category/ChannelsContainer/index.tsx

```diff
@@ -1,50 +1,54 @@
 import { Fragment, RefObject, forwardRef } from 'react';
-import { ChannelsWrapper, ThreadsWrapper } from '@components/Sidebar/ChannelsList/elements';
+import * as Styles from '@components/Sidebar/ChannelsSidebar/styles';
 import { useStoreState } from '@state';
 import { Category as ICategory } from '@graphql/graphql';
-import { Spine } from '@components/Shared/Icons/Spine';
+import { ThreadSpine } from '@components/Shared/Icons/ThreadSpine';
+import { useAppRouter } from '@hooks/useAppRouter';
 import { Channel } from './Channel';
 
 interface ChannelsProps {
-  currentChannelID: string;
-  currentThreadID: string;
   isCategoryOpen: boolean;
+
   category: ICategory;
+
   currentChannelRef: RefObject<HTMLAnchorElement>;
 }
 /** Component that handles rendering text and thread channels */
 export const ChannelsContainer = forwardRef<HTMLDivElement, ChannelsProps>(
-  ({ currentChannelID, isCategoryOpen, category, currentChannelRef, currentThreadID }, ref) => {
+  ({ isCategoryOpen, category, currentChannelRef }, ref) => {
     const channels = useStoreState(state => state.guild.channels!);
+    const { threadId, channelId } = useAppRouter();
 
     return (
       <div ref={ref}>
         {channels
           .filter(c => c.category?.id === category?.id)
           .map(channel => (
             <Fragment key={channel.id}>
-              <ChannelsWrapper draggable={false} className="channels-wrapper">
+              <Styles.ChannelsWrapper draggable={false}>
                 <Channel
                   channel={channel}
-                  isActive={!currentThreadID && channel.id === currentChannelID}
+                  isActive={channel.id === channelId}
                   isCategoryOpen={isCategoryOpen}
                   ref={currentChannelRef}
+                  channelHasActiveThread={!!threadId && channel.id === channelId}
                 />
-              </ChannelsWrapper>
+              </Styles.ChannelsWrapper>
 
               {channel.threads?.map(thread => (
                 <Fragment key={thread.id}>
-                  {currentThreadID === thread.id && (
-                    <ThreadsWrapper className="thread-wrapper">
-                      <Spine />
+                  {threadId === thread.id && (
+                    <Styles.ThreadsWrapper>
+                      {thread.id === threadId && <ThreadSpine />}
                       <Channel
                         isThread
                         channel={thread}
-                        isActive={thread.id === currentThreadID}
+                        isActive={thread.id === threadId}
                         isCategoryOpen={isCategoryOpen}
                         ref={currentChannelRef}
+                        channelHasActiveThread={!!threadId && channel.id === channelId}
```

Comment:
```suggestion
                        channelHasActiveThread={threadId && channel.id === channelId}
```

or if it really needs to be a boolean
```suggestion
                        channelHasActiveThread={Boolean(threadId) && channel.id === channelId}
```

Reply:
Throws type error without double bang